src/tests/dictionaryConcepts/container/EditConcept.test.jsx

```diff
@@ -0,0 +1,124 @@
+import Router from 'react-mock-router';
+import {
+  EditConcept,
+  mapStateToProps,
+} from '../../../components/dictionaryConcepts/containers/EditConcept';
+import { newConcept, existingConcept } from '../../__mocks__/concepts';
+
+jest.mock('uuid/v4', () => jest.fn(() => 1234));
+jest.mock('react-notify-toast');
+
+global.localStorage = {
+  getItem: () => '/',
+};
+
+describe('Test suite for dictionary concepts components', () => {
+  const props = {
+    match: {
+      params: {
+        conceptType: 'diagnosis',
+        collectionName: 'dev-col',
+        type: 'users',
+        typeName: 'emmabaye',
+        language: 'en',
+      },
+    },
+    history: {
+      push: jest.fn(),
+    },
+    createNewName: jest.fn(),
+    addNewDescription: jest.fn(),
+    clearSelections: jest.fn(),
+    fetchExistingConcept: jest.fn(),
+    clearPreviousConcept: jest.fn(),
+    createNewNameForEditConcept: jest.fn(),
+    removeDescriptionForEditConcept: jest.fn(),
+    addDescriptionForEditConcept: jest.fn(),
+    removeNameForEditConcept: jest.fn(),
+    updateConcept: jest.fn(),
+    newName: ['1'],
+    description: ['1'],
+    existingConcept: {
+      names: [{
+        uuid: '1234',
+      }],
+      descriptions: [{
+        uuid: '1234',
+      }],
+    },
+  };
+  it('should render without breaking', () => {
+    const wrapper = mount(<Router>
+      <EditConcept {...props} />
+    </Router>);
+    expect(wrapper.find('h3').text()).toEqual(': Edit a diagnosis Concept ');
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should handle form completion and submission with invalid/incomplete data', () => {
+    const wrapper = mount(<Router>
+      <EditConcept {...props} />
+    </Router>);
+    wrapper.find('#toggleUUID').simulate('click');
+    const event = { target: { name: 'id', value: '12345ft-007' } };
+    wrapper.find('#uuid').simulate('change', event);
+    wrapper.find('#createConceptForm').simulate('submit', {
+      preventDefault: () => {},
+    });
+    const conceptName = { target: { name: 'name', value: 'test concept' } };
+    wrapper.find('#concept-name').simulate('change', conceptName);
+    const conceptDatatype = { target: { name: 'datatype', value: '' } };
+    wrapper.find('#datatype').simulate('change', conceptDatatype);
+    wrapper.find('#toggleUUID').simulate('click');
+    const uuid = { target: { name: 'id', value: '12345ft-007#' } };
+    wrapper.find('#uuid').simulate('change', uuid);
+    const conceptDescription = {
+      target: { name: 'description', value: 'test concept description' },
+    };
+    wrapper.find('#concept-description').simulate('change', conceptDescription);
+    wrapper.find('#remove-description').simulate('click');
+    wrapper.find('#createConceptForm').simulate('submit', {
+      preventDefault: () => {},
+    });
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should handle form completion and submission with correct data', () => {
```

Comment:
@emmabaye, I also think, testing for form input data completion should be done differently from the form submission. If the two are critical to the form functionality and application or rename the test. Also, separate the tests for clicking the remove button. Why? this whole test will fail in case the remove on click event fails which also renders the onChange events as failed. Test for what you expect when you click. E.g. test that something that was existing is now gone.

Reply:
Otherwise, good stuff.

Reply:
Thanks for your feedback. However, for a form to be submitted, it has to be complete first, the reason for the "change" simulations before the form is submitted. The form is made to submit if all the necessary information is provided. A concept might not necessarily have names and descriptions as these are optional.

Reply:
Updated